fe/fe-core/src/main/java/com/starrocks/sql/optimizer/validate/TypeChecker.java

```diff
@@ -150,20 +150,24 @@ private void checkFuncCall(Map<ColumnRefOperator, CallOperator> functionCalls) {
         private void checkAggCall(Map<ColumnRefOperator, CallOperator> aggregations, AggType aggType, boolean isSplit,
                                   int singleDistinctFunctionPos) {
             int index = -1;
+            boolean hasSingleDistinct = singleDistinctFunctionPos != - 1;
             for (Map.Entry<ColumnRefOperator, CallOperator> entry : aggregations.entrySet()) {
                 index++;
                 ColumnRefOperator outputCol = entry.getKey();
                 CallOperator aggCall = entry.getValue();
                 boolean isMergeAggFn = false;
-                if (aggType.isGlobal() || (aggType.isLocal() && !isSplit)) {
-                    if (singleDistinctFunctionPos != -1 && index != singleDistinctFunctionPos) {
+                switch (aggType) {
+                    case LOCAL:
+                        isMergeAggFn = isSplit && hasSingleDistinct && (index != singleDistinctFunctionPos);
```

Comment:
LOCAL should always be update mode
1.  LOGAL AGG -> RESULT update mode
2.  LOGAL AGG -> AGG-> ... > RESULT update mode
GLOBAL should be merge mode excluding one-phase global agg.
2.  AGG->...> GLOBAL->RESULT  merge mode
3.  Shuffle-> GLOBAL->RESULT update mode
DISTINCT_GLOBAL and DISTINCT_LOCAL  should always be merge mode.
AGG->...->DISTINCT_GLOBAL/DISTINCT_LOCAL->...->  merge mode

Reply:
Because we may split the top global agg into a two phase agg, so local may also has update mode. Like
```
LOCAL -> DSITINCT_GLOBAL -> GLOBAL
```
to
```
LOCAL -> DSITINCT_GLOBAL -> LOCAL -> GLOBAL
```